### PR TITLE
feat: add pod controller and pod handler

### PR DESCRIPTION
**Pull request by OrangeBao**

### CODE REVIEW

File: pkg/knode-manager/utils/conversion.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+func recoverSelectors(pod *corev1.Pod, cns *ClustersNodeSelection) {
+	if cns != nil {
+		pod.Spec.NodeSelector = cns.NodeSelector
+		pod.Spec.Tolerations = cns.Tolerations
+		if pod.Spec.Affinity == nil {
+			pod.Spec.Affinity = cns.Affinity
+		} else {
+			if cns.Affinity != nil && cns.Affinity.NodeAffinity != nil {
```
Comment (wuyingjun-lucky) [inline pkg/knode-manager/utils/conversion.go:120]: the logic seems too complex

File: pkg/knode-manager/utils/expansion/expand.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package expansion
+
```
Comment (wuyingjun-lucky) [inline pkg/knode-manager/utils/expansion/expand.go:2]: what does expansion use for

File: pkg/knode-manager/adapters/k8s/help.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
+package k8sadapter
+
```
Comment (duanmengkk) [inline pkg/knode-manager/adapters/k8s/help.go:2]: help.go or helps.go ? I did not see any code file named help.go

File: pkg/knode-manager/adapters/k8s/help.go [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+			secretNames = append(secretNames, v.Secret.SecretName)
+
+		case v.CephFS != nil:
+			klog.Infof("pod %s depends on secret %s", pod.Name, v.CephFS.SecretRef.Name)
+			secretNames = append(secretNames, v.CephFS.SecretRef.Name)
+		case v.Cinder != nil:
+			klog.Infof("pod %s depends on secret %s", pod.Name, v.Cinder.SecretRef.Name)
+			secretNames = append(secretNames, v.Cinder.SecretRef.Name)
+		case v.RBD != nil:
+			klog.Infof("pod %s depends on secret %s", pod.Name, v.RBD.SecretRef.Name)
```
Comment (duanmengkk) [inline pkg/knode-manager/adapters/k8s/help.go:28]: We can leave out this part of the logic for now and let the developers from the open-source community who join later add it

File: pkg/knode-manager/adapters/k8s/help.go (hunk 3 above)
Comment (OrangeBao) [reply pkg/knode-manager/adapters/k8s/help.go:2]: chang name to util.go?

File: pkg/knode-manager/adapters/k8s/help.go [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+		RestartPolicy == corev1.RestartPolicyNever
+}
+
+// nolint:unused
+func nodeCustomLabel(node *corev1.Node, label string) {
+	nodelabel := strings.Split(label, ":")
+	if len(nodelabel) == 2 {
+		node.ObjectMeta.Labels[strings.TrimSpace(nodelabel[0])] = strings.TrimSpace(nodelabel[1])
+	}
+}
```
Comment (duanmengkk) [inline pkg/knode-manager/adapters/k8s/help.go:97]: what does the package name adapters means

File: pkg/knode-manager/adapters/k8s/pod.go [hunk 6]
```diff
@@ -2,43 +2,606 @@ package k8sadapter
...
+			"update serviceAccount [%v] err: [%v]]",
+			sa, err)
+		return err
+	}
+	return nil
 }
 
 func (p *PodAdapter) Create(ctx context.Context, pod *corev1.Pod) error {
-	return network.ErrNotImplemented
+	if pod.Namespace == "kube-system" {
```
Comment (duanmengkk) [inline pkg/knode-manager/adapters/k8s/pod.go:420]: magic value

File: pkg/knode-manager/utils/expansion/expand.go (hunk 2 above)
Comment (OrangeBao) [reply pkg/knode-manager/utils/expansion/expand.go:2]: Parsing variable templates, like $(varx)

File: pkg/knode-manager/adapters/k8s/help.go (hunk 5 above)
Comment (OrangeBao) [reply pkg/knode-manager/adapters/k8s/help.go:97]: interface implementation

File: pkg/knode-manager/adapters/k8s/help.go (hunk 4 above)
Comment (OrangeBao) [reply pkg/knode-manager/adapters/k8s/help.go:28]: For now, let's do this

File: pkg/knode-manager/utils/util.go [hunk 7]
```diff
@@ -1 +1,10 @@
 package utils
+
```
Comment (wuyingjun-lucky) [inline pkg/knode-manager/utils/util.go:2]: utils/util.go ? seems strange

File: pkg/knode-manager/adapters/k8s/help.go (hunk 4 above)
Comment (duanmengkk) [reply pkg/knode-manager/adapters/k8s/help.go:28]: we do not need it now

File: pkg/knode-manager/utils/util.go (hunk 7 above)
Comment (OrangeBao) [reply pkg/knode-manager/utils/util.go:2]: refer to this:
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/config/config.go
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/coverage/coverage.go

xxx/xxx.go

Comment (OrangeBao) [reply pkg/knode-manager/utils/util.go:2]: sigs.k8s.io/kustomize/kyaml/internal/forked/github.com/qri-io/starlib/util/util.go

util/util.go

Comment (wuyingjun-lucky) [reply pkg/knode-manager/utils/util.go:2]: They are not same.
 when they have `xxx/xxx.go`， there are only `xxx.go`  under the `xxx` package . But we have others

Comment (OrangeBao) [reply pkg/knode-manager/utils/util.go:2]: this is not the key issue, don`t worry.

**Review by wangyizhi1 — APPROVED** [review]

**Pull request merged**